src/main/webapp/app/core/course/manage/quick-actions/add-exercise-modal/add-exercise-modal.component.html

```diff
@@ -0,0 +1,71 @@
+                </div>
+                <div class="col-4">
+                    <jhi-programming-create-button [course]="course" [translationKey]="'global.generic.create'" />
+                </div>
+                <div class="col-4">
+                    <jhi-programming-import-button [course]="course" [translationKey]="'entity.action.to-import'" />
+                </div>
+            </div>
+
+            <div class="row align-items-center g-0">
+                <div class="col-4 pe-0">
+                    <strong jhiTranslate="global.menu.entities.exerciseTypes.quiz"></strong>
+                </div>
+                <div class="col-4">
+                    <jhi-quiz-create-button [course]="course" [translationKey]="'global.generic.create'" />
+                </div>
+                <div class="col-4">
+                    <jhi-quiz-import-button [course]="course" [translationKey]="'entity.action.to-import'" />
+                </div>
+            </div>
+
+            <div class="row align-items-center g-0">
+                <div class="col-4 pe-0">
+                    <strong jhiTranslate="global.menu.entities.exerciseTypes.modeling"></strong>
+                </div>
+                <div class="col-4">
+                    <jhi-exercise-create-button [course]="course" [exerciseType]="ExerciseType.MODELING" [translationKey]="'global.generic.create'" />
+                </div>
+                <div class="col-4">
+                    <jhi-exercise-import-button [course]="course" [exerciseType]="ExerciseType.MODELING" [translationKey]="'entity.action.to-import'" />
+                </div>
+            </div>
+
+            <div class="row align-items-center g-0">
+                <div class="col-4 pe-0">
+                    <strong jhiTranslate="global.menu.entities.exerciseTypes.text"></strong>
+                </div>
+                <div class="col-4">
+                    <jhi-exercise-create-button [course]="course" [exerciseType]="ExerciseType.TEXT" [translationKey]="'global.generic.create'" />
+                </div>
+                <div class="col-4">
+                    <jhi-exercise-import-button [course]="course" [exerciseType]="ExerciseType.TEXT" [translationKey]="'entity.action.to-import'" />
+                </div>
+            </div>
+
+            <div class="row align-items-center g-0">
+                <div class="col-4 pe-0">
+                    <strong jhiTranslate="global.menu.entities.exerciseTypes.fileUpload"></strong>
+                </div>
+                <div class="col-4">
+                    <jhi-exercise-create-button [course]="course" [exerciseType]="ExerciseType.FILE_UPLOAD" [translationKey]="'global.generic.create'" />
+                </div>
+                <div class="col-4">
+                    <jhi-exercise-import-button [course]="course" [exerciseType]="ExerciseType.FILE_UPLOAD" [translationKey]="'entity.action.to-import'" />
+                </div>
+            </div>
```

Comment:
In my opinion, this is not a good approach, I think we can solve this more data driven.

Please solve it with less code duplication. This might be achievable by using an `ng-template`, I am not quite sure how to pass a component there.
Defining the component can be solved by defining a directive (e.g. have a look at the directive `jhiExerciseDetail`)